Add AUTHORS file and update script

tunniclm opened the pull request:
tools/update-authors.sh is based on the same file originating from the
https://github.com/nodejs/node repository. The LICENSE file has been
updated to include the license for this file.

jordonias committed:
add Gitter link to Readme.md

Commit:
Merge pull request #2848 from jordonias/master

add Gitter link to Readme.md

jasnell commented:
LGTM

hacksparrow commented on LICENSE:
```diff
@@ -22,3 +26,29 @@ IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+update-authors.sh from https://github.com/nodejs/node:
+
+"""
+Copyright Node.js contributors. All rights reserved.
```
> Copyright Node.js contributors?

tunniclm replied:
@hacksparrow The script is covered by this license. I did the simple thing and just included it. Maybe we can do something more condensed to cover any files that originate externally and have a line for <file> <copyright> <license type> as a more lightweight summary though.

dougwilson commented:
Hi @tunniclm, thank you for your pull request!

So, it would be really nice if you can make the script run on Windows machine, for Windows-centric contributors like myself. A shell script means that only a subset of contributors would be capable of maintaining the file. I suggest writing the script in Node.js itself, which would guarantee that all contributors could run the script, regardless of their operating system of choice.

Also, please try to clean up the file a bit, as there are some people listed multiple times. Of particular note is TJ himself is in there multiple times.

As far as the `LICENSE` file update goes, the current update is bound to cause confusion. This is because how `package.json` is setup, the `tools/` directory you added is not published to `npm`, but yet the modified `LICENSE` is going to be published to `npm`. I would personally recommend either moving the `LICENSE` file into the `tools/` directory, or even better, just inline the license in a comment block in the script itself.

tunniclm commented:
@dougwilson All good points. I have written a node.js script instead of the shell/perl script. It still requires the git command-line client to be on the path to run.

I spent a little time enhancing the duplicate removal in the script, however, I don't want to take it beyond same-email address  as it would open up the possibility of mistakes. For example, if you see "John Doe john@gmail.com" make a commit then later "John Doe <john@doe.net" then you might  say these were the same person, but you wouldn't really be sure.

I will update the PR shortly.

tunniclm committed:
Add AUTHORS file and update script

tools/update_authors.js reads the git history (using the git command
line client) and processes the output to remove duplicate entries
(keyed from the user's email address).

The resulting list of authors is written to a file AUTHORS in the
current working directory. Author entries are in order of first
commit but use the most recent name for the given email address.

tunniclm commented:
@dougwilson Oops, forgot to tag you to let you know I updated this pr. Do the changes address your concerns? Thanks!

Fishrock123 commented:
Do we actually need this?

tunniclm commented:
It sounds like this is not needed, so closing it out.
